doc/SONIC_Test_Ingress_Discard_All_HLD.md

```diff
@@ -0,0 +1,606 @@
+### [DRAFT, UNDER DEVELOPMENT]
+
+- [Overview](#overview)
+    - [Scope](#scope)
+    - [Related DUT CLI commands](#related-dut-cli-commands)
+- [General test flow](#general-test-flow)
+- [Run test](#run-test)
+- [Test cases](#test-cases)
+    - [Test case #1](#test-case-1)
+    - [Test case #2](#test-case-2)
+    - [Test case #3](#test-case-3)
+    - [Test case #4](#test-case-4)
+    - [Test case #5](#test-case-5)
+    - [Test case #6](#test-case-6)
+    - [Test case #7](#test-case-7)
+    - [Test case #8](#test-case-8)
+    - [Test case #9](#test-case-9)
+    - [Test case #10](#test-case-10)
+    - [Test case #11](#test-case-11)
+    - [Test case #12](#test-case-12)
+    - [Test case #13](#test-case-13)
+    - [Test case #14](#test-case-14)
+    - [Test case #15](#test-case-15)
+    - [Test case #16](#test-case-16)
+
+#### [Note]
+Proposal to implement more extendent test cases on phase 2 when https://github.com/Azure/SONiC/blob/master/doc/drop_counters/drop_counters_HLD.md will be implemented.
+
+#### Overview
+The purpose is to test "ingress_discard_all" counter triggers on receiving specific packets.
+The "ingress_discard_all" counter counts all discard events. This counter counts concurrently with other discard counters.
+The test assumes all necessary configuration are already pre-configured on the SONIC switch before test runs.
+Destination IP address of the injected packet must be routable to ensure packet should be routed but was dropped.
+
+#### Scope
+The purpose of the test is testing of "ingress_discard_all" counter triggering on SONIC system, making sure that specific traffic drops correctly, according to sent packet and configured packet discards.
+Supported topologies:
+```
+t0
+t1
+t1-lag
+ptf32
+```
+
+#### Related DUT CLI commands
+Command to check "ingress_discard_all" counter value:
+```show interfaces counters```
+
+Check field:
+```RX_DRP```
+
+#### General test flow
+- Select two pairs of PTF and DUT ports considering topology: PTF_PORT[1] <--->DUT_PORT[1], PTF_PORT[2] <--->DUT_PORT[2]
+- Clear counters. Use CLI command "sonic-clear counters"
```

Comment:
should you cover also router ports? lag ports? more?

Reply:
As a general comment I agree with Liat, this design should include how the test chooses the ports to test according to supported typologies,
I believe according to what topology configuration is you should test:
1. port which is a member of L3 vlan (T0)
2. port which is a member of L3 LAG(T0, T1-LAG)
3. L3 port (T1, PTF32)

Reply:
Added section that describes that test cases will use available port groups depending on the topology. For example, if topology has LAG, RIFs, VLANS test will send traffic on these ports, it will be done as separate test case.